src/core/on-chain/index.ts

```diff
@@ -378,20 +356,12 @@ export const OnChainMixin = (superclass) =>
       // this would avoid a communication to the server
       // every time you want to show a QR code.
 
-      let address
-
-      const { lnd, pubkey } = getActiveOnchainLnd()
+      const address = await this.bitcoindWalletClient.getNewAddress({
+        address_type: "bech32", // TODO confirm
+      })
 
-      try {
-        ;({ address } = await createChainAddress({
-          lnd,
-          format: "p2wpkh",
-        }))
-      } catch (err) {
-        const error = `error getting on chain address`
-        this.logger.error({ err }, error)
-        throw new LoggedError(error)
-      }
+      // TODO pubkey should be known already
+      const { pubkey } = await this.bitcoindWalletClient.getAddressInfo({ address })
```

Comment:
I am not sure if we need this pubkey, the previous pubkey is the pubkey of lnd node that is needed to get chain data in the cron and trigger.